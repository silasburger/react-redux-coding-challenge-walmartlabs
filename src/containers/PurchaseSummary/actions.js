import { getPurchaseData } from './mocks/purchaseDataMock';
import {
  RECEIVED_PRICING_DATA,
  RECEIVED_ITEM_DATA,
  APPLY_DISCOUNT
} from './constants';

export function fetchPurchaseData() {
  return function(dispatch) {
    getPurchaseData()
      .then(response => {
        dispatch(recievedPricingData(response.pricing));
        dispatch(recievedItemData(response.itemDetails))
      })
      .catch(err => {
        console.error(err);
      });
  };
}

export function recievedPricingData(data) {
  return { type: RECEIVED_PRICING_DATA, data };
}

export function recievedItemData(data) {
  return { type: RECEIVED_ITEM_DATA, data };
}

export function applyDiscount(discountCode) {
  return { type: APPLY_DISCOUNT, discountCode };
}
