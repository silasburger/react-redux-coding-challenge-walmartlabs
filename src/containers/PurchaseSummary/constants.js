export const RECEIVED_PRICING_DATA = 'RECEIVED_PRICING_DATA';  
export const RECEIVED_ITEM_DATA = 'RECEIVED_ITEM_DATA';  
export const APPLY_DISCOUNT = 'APPLY_DISCOUNT';  