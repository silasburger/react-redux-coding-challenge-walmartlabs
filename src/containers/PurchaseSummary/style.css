.purchase-summary {
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  float: right;
  width: 40%;
  border: solid 1px #ccc;
  padding: 20px;
  border-radius: 7px;
}
