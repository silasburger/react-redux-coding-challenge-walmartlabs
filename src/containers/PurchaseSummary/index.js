import React, { Component } from 'react';
import { connect } from 'react-redux';
import Row from '../../components/Row';
import ExpandableItem from '../../components/ExpandableItem';
import DiscountForm from '../../components/DiscountForm';
import ItemDetails from '../../components/ItemDetails';
import ToolTip from '../../components/ToolTip';
import { applyDiscount, fetchPurchaseData } from './actions';
import './style.css';

class PurchaseSummary extends Component {
  async componentDidMount() {
    try {
      await this.props.fetchPurchaseData();
    } catch (err) {
      console.error(err);
    }
  }

  render() {
    return (
      <div className="purchase-summary">
          <Row title="Subtotal" figure={this.props.pricing.subtotal} />
          <Row
            title="Pickup savings"
            figure={this.props.pricing.savings}
            less={true}
            toolTip={
              <ToolTip
                title="Pickup savings"
                desc="Picking up your order in the store helps cut costs, and we pass the savings on to you."
              />
            }
          />
          {this.props.pricing.discount > 0 ? (
            <Row
              title="Discount"
              figure={this.props.pricing.discount}
              less={true}
            />
          ) : null}
          <Row title="Est. taxes and fees" figure={this.props.pricing.tax}>
            <div>(Based on {this.props.pricing.zip})</div>
          </Row>
          <Row
            total={true}
            title="Est. total"
            figure={this.props.pricing.total}
          />
        <ExpandableItem
          openPrefix={'See'}
          closePrefix={'Hide'}
          title={'item details'}
        >
          <ItemDetails {...this.props.item} />
        </ExpandableItem>
        <ExpandableItem
          openPrefix={'Apply'}
          closePrefix={'Hide'}
          title={'promo code'}
        >
          <DiscountForm applyDiscount={this.props.applyDiscount} />
        </ExpandableItem>
      </div>
    );
  }
}

function mapStateToProps(state, ownProps) {
  return {
    pricing: state.pricing,
    item: state.item
  };
}

export default connect(
  mapStateToProps,
  { applyDiscount, fetchPurchaseData }
)(PurchaseSummary);
