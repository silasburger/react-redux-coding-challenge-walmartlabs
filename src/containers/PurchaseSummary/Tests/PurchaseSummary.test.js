import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import PurchaseSummary from '../index';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
describe('purchase summary container', () => {
  it('matches snapshot', function() {
    let wrapper = shallow(<PurchaseSummary />);
    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });

  it('renders without crashing', function() {
    let wrapper = shallow(<PurchaseSummary />);
    console.log(wrapper.debug());
  });
});
