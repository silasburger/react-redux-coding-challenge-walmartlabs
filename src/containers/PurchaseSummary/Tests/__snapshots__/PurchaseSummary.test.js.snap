// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`purchase summary container matches snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
